cmake: arm64: Bail out if the old SDK is used

carlocaione opened the pull request:
After the ARM/ARM64 split the SDK 0.12.4 is needed to be able to compile
the ARM64 arch code. Bail out if an old SDK is detected at compile time.

Closes: #34046

tejlmand commented on cmake/toolchain/zephyr/target.cmake:
```diff
@@ -15,4 +15,10 @@ else()
+endif()
+
+# The ARM/ARM64 split requires at least the 0.12.4 SDK
+if(("${ARCH}" STREQUAL "arm64") AND (${SDK_VERSION} VERSION_LESS_EQUAL 0.12.3))
```
actually this should go here:
https://github.com/zephyrproject-rtos/zephyr/blob/55771d36a7a7a452b962c81bdef832fcb7be570b/cmake/verify-toolchain.cmake#L18-L19

and then take the form:
```
if("${ARCH}" STREQUAL "arm64")
  set(TOOLCHAIN_ZEPHYR_MINIMUM_REQUIRED_VERSION 0.12.4)
else()
  set(TOOLCHAIN_ZEPHYR_MINIMUM_REQUIRED_VERSION 0.11.3)
endif()
```

and then we should update:
https://github.com/zephyrproject-rtos/zephyr/blob/55771d36a7a7a452b962c81bdef832fcb7be570b/cmake/verify-toolchain.cmake#L70
and 
https://github.com/zephyrproject-rtos/zephyr/blob/55771d36a7a7a452b962c81bdef832fcb7be570b/cmake/verify-toolchain.cmake#L75
to include `REQUIRED`.

That will give regular CMake error in the form of:
```
CMake Error at /projects/github/ncs/zephyr/cmake/verify-toolchain.cmake:75 (find_package):
  Could not find a configuration file for package "Zephyr-sdk" that is
  compatible with requested version "0.12.4".

  The following configuration files were considered but not accepted:

    /opt/zephyr-sdk-0.12.3/cmake/Zephyr-sdkConfig.cmake, version: 0.12.3
```

Only thing to decide is whether it's time to remove support for `0.11.1` and `0.11.2`.
When adding `REQUIRED`, then it will no longer be possible to point to a `0.11.1` or `0.11.2`.
@galak thoughts ?

carlocaione replied:
@galak ping, so that I can push a new revision

tejlmand replied:
@abrodkin @evgeniy-paltsev could you comment.

tejlmand replied:
Discussed at Toolchain WG.
Will update to minimum 0.12.

galak commented:
@carlocaione any updates on this?

carlocaione commented:
> @carlocaione any updates on this?

Ups, this fell through the cracks. I'll take a look ASAP again.

carlocaione committed:
cmake: arm64: Bail out if the old SDK is used

After the ARM/ARM64 split the SDK 0.12.4 is needed to be able to compile
the ARM64 arch code. Bail out if an old SDK is detected at compile time.

Also change the minimal supported Zephyr SDK to v0.12, as agreed in
Toolchain WG.  Main reason is added compiler architectures and important
inclusion of fixes, for example qemu and arm fixes for veneers (TFM)

Signed-off-by: Carlo Caione <ccaione@baylibre.com>

carlocaione commented:
@galak @tejlmand I'm not really familiar with CMake so you probably want to take a look and deprecate <0.12

tejlmand commented on cmake/verify-toolchain.cmake:
```diff
@@ -16,7 +16,11 @@
+if("${ARCH}" STREQUAL "arm64")
+  set(TOOLCHAIN_ZEPHYR_MINIMUM_REQUIRED_VERSION 0.12.4)
+else()
+  set(TOOLCHAIN_ZEPHYR_MINIMUM_REQUIRED_VERSION 0.11.3)
```
As per https://github.com/zephyrproject-rtos/zephyr/pull/34064#discussion_r633641075

please change this directly to `0.12`
```suggestion
  set(TOOLCHAIN_ZEPHYR_MINIMUM_REQUIRED_VERSION 0.12)
```

tejlmand reviewed (changes requested):
Looks good.
Please directly go for 0.12.

MaureenHelm commented on cmake/verify-toolchain.cmake:
```diff
@@ -16,7 +16,11 @@
+if("${ARCH}" STREQUAL "arm64")
+  set(TOOLCHAIN_ZEPHYR_MINIMUM_REQUIRED_VERSION 0.12.4)
+else()
+  set(TOOLCHAIN_ZEPHYR_MINIMUM_REQUIRED_VERSION 0.12)
```
Can you update the commit message to explain why we're also increasing the SDK version for other arches too?

tejlmand replied:
@carlocaione you can just add:
```
Minimal Zephyr SDK v0.12, as agreed in Toolchain WG.
Main reason is added compiler architectures and important inclusion of fixes,
for example qemu and arm fixes for veneers (TFM)
```

tejlmand replied:
@MaureenHelm done

MaureenHelm replied:
Thank you

yashi commented:
`REQUIRED` cause `find_package()` to bail out at that point if it failed to find suitable SDK. That means that we don't execute the following if block nor print nice message about what we have installed and where to look for.  Is it time to remove the if block as discussed above? or do we want to keep it to inform users where to look for?  The URL is also broken and IMHO we should point to the Getting Started Guide instead of direct link to the minimum required SDK.
I'll create an issue if we need to.  Let me know what you guys think.
